core/src/io/anuke/mindustry/core/Control.java

```diff
@@ -176,6 +176,16 @@ public Control(){
         Events.on(ZoneConfigureCompleteEvent.class, e -> {
             ui.hudfrag.showToast(Core.bundle.format("zone.config.complete", e.zone.configureWave));
         });
+
+        Events.on(WaveEvent.class, e-> {
```

Comment:
`Control` is only created on the client, so this code will never run on a server. This should be in `Logic`.